concent_api/core/models.py

```diff
@@ -73,3 +76,167 @@ class MessageAuth(Model):
     message              = OneToOneField(StoredMessage, related_name = 'auth')
     provider_public_key  = Base64Field(max_length=64)
     requestor_public_key = Base64Field(max_length=64)
+
+
+class ClientManager(Manager):
+
+    def get_or_create_full_clean(self, public_key):
+        """
+        Returns Model instance.
+        Does the same as get_or_create method, but also performs full_clean() on newly created instance.
+        """
+        try:
+            instance = self.get(public_key = base64.b64encode(public_key))
+        except self.model.DoesNotExist:
+            instance = self.model(
+                public_key_bytes = public_key
+            )
+            instance.full_clean()
+            instance.save()
+        return instance
+
+
+class Client(models.Model):
+    """
+    Represents Concent client, identified by public key.
+    """
+
+    objects = ClientManager()
+
+    public_key = Base64Field(max_length = 64, unique = True)
+
+
+class Subtask(models.Model):
+    """
+    Represents subtasks states.
+    """
+
+    class SubtaskState(ChoiceEnum):
+        FORCING_REPORT          = 'forcing_report'
+        REPORTED                = 'reported'
+        FORCING_RESULT_TRANSFER = 'forcing_result_transfer'
+        RESULT_UPLOADED         = 'result_uploaded'
+        FORCING_ACCEPTANCE      = 'forcing_acceptance'
+        REJECTED                = 'rejected'
+        ADDITIONAL_VERIFICATION = 'additional_verification'
+        ACCEPTED                = 'accepted'
+        FAILED                  = 'failed'
+
+    # Defines Subtask model active states
+    SUBTASK_ACTIVE_STATES = [
+        SubtaskState.FORCING_REPORT.name,  # pylint: disable=no-member
```

Comment:
There is something funky going on with those enums, I think we shouldn't use `.name` or `.value`, it needs some more investigation why this is happening.

Reply:
I agree it looks weird but surprisingly works correctly this way and only this way. Yesterday I have tried to rebuild to use enums directly, without `.value` or `.name`, but each time I have ended in having different issues either in comparison, querying or storing to database.

Reply:
After discussion it looks like the issue is really there, I will update this piece.

Reply:
:+1: